contrib/mesos/pkg/offers/offers.go

```diff
@@ -441,7 +441,7 @@ func (s *offerStorage) ageOffers() {
 }
 
 func (s *offerStorage) nextListener() *offerListener {
-	obj := s.listeners.Pop()
+	obj := s.listeners.Pop(nil)
```

Comment:
I didn't like the multiple funcs when a single func does the job just fine. I don't see anything wrong with an API that accepts an optional arg. I can add docs to clarify the intended usage.